### PR TITLE
fix: revalidate keys page

### PR DESCRIPTION
**Pull request by MichaelUnkey**
## What does this PR do?

eng-1080

Fixes # (issue)

*If there is not an issue for this, please create one first. This is used to tracking purposes and also helps use understand why this PR exists*

## Type of change

* [x] Bug fix (non-breaking change which fixes an issue)
* [ ] Chore (refactoring code, technical debt, workflow improvements)
* [ ] Enhancement (small improvements)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

## How should this be tested?

Check to see if key is present after deletion. And no errors are found

### Required

* [x] Filled out the "How to test" section in this PR
* [x] Read [Contributing Guide](./CONTRIBUTING.md)
* [x] Self-reviewed my own code
* [x] Commented on my code in hard-to-understand areas
* [x] Ran `pnpm build`
* [x] Ran `pnpm fmt`
* [x] Checked for warnings, there are none
* [x] Removed all `console.logs`
* [x] Merged the latest changes from main onto my branch with `git pull origin main`
* [x] My changes don't cause any responsiveness issues

### Appreciated

* [ ] If a UI change was made: Added a screen recording or screenshots to this PR
* [ ] Updated the Unkey Docs if changes were necessary

<sub><a href="https://huly.app/guest/unkey?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NjgzMGMwZjJhNmFiOWVlZjU1OWFhMjMiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctamFtZXMtdW5rZXktNjY2YWYwNzktNjliNzJiMzYyNi05ZGM2MGUiLCJwcm9kdWN0SWQiOiIifQ.XU4Bw9DMVFjuk88POVKFhwBVTim87O_doZ7D4z7kxrU">Huly&reg;: <b>ENG-1834</b></a></sub>

### CODE REVIEW

File: apps/dashboard/app/(app)/keys/[keyAuthId]/new/page.tsx [hunk 1]
```diff
@@ -17,7 +18,7 @@ export default async function ApiPage(props: { params: { keyAuthId: string } })
   if (!keyAuth || keyAuth.workspace.tenantId !== tenantId) {
     return notFound();
   }
-
+  revalidate(`/keys/${props.params.keyAuthId}`);
```
Comment (chronark) [inline apps/dashboard/app/(app)/keys/[keyAuthId]/new/page.tsx:21]: this revalidates on page render, but it should revalidate when a mutation finishes
you need to call this in the mutation

File: apps/dashboard/app/(app)/keys/[keyAuthId]/[keyId]/settings/delete-key.tsx [hunk 2]
```diff
@@ -27,10 +27,14 @@ type Props = {
 
 export const DeleteKey: React.FC<Props> = ({ apiKey }) => {
   const router = useRouter();
+  const searchParams = useSearchParams();
```
Comment (chronark) [inline apps/dashboard/app/(app)/keys/[keyAuthId]/[keyId]/settings/delete-key.tsx:30]: let's pass this as a prop, just like the apiKey please

**Review by chronark — CHANGES_REQUESTED** [review]

**Review by perkinsjr — CHANGES_REQUESTED** [review]
this is a test

File: apps/dashboard/app/(app)/apis/[apiId]/page.tsx [hunk 3]
```diff
@@ -17,7 +17,6 @@ import { redirect } from "next/navigation";
 import { type Interval, IntervalSelect } from "./select";
 
 export const dynamic = "force-dynamic";
-export const runtime = "edge";
```
Comment (chronark) [inline apps/dashboard/app/(app)/apis/[apiId]/page.tsx:20]: why is this necessary?

File: apps/dashboard/app/(app)/keys/[keyAuthId]/[keyId]/settings/delete-key.tsx [hunk 4]
```diff
@@ -17,22 +17,27 @@ import {
...
-export const DeleteKey: React.FC<Props> = ({ apiKey }) => {
+export const DeleteKey: React.FC<Props> = ({ apiKey, keyAuthId }) => {
   const router = useRouter();
   const [open, setOpen] = useState(false);
 
   const deleteKey = trpc.key.delete.useMutation({
     onSuccess() {
+      revalidate(`/keys/${keyAuthId}`);
+      revalidate(`/keys/${keyAuthId}/keys`);
+      revalidate("/apis");
```
Comment (chronark) [inline apps/dashboard/app/(app)/keys/[keyAuthId]/[keyId]/settings/delete-key.tsx:37]: why is this necesary?

File: apps/dashboard/app/(app)/keys/[keyAuthId]/[keyId]/settings/delete-key.tsx [hunk 5]
```diff
@@ -17,22 +17,27 @@ import {
...
 };
 
-export const DeleteKey: React.FC<Props> = ({ apiKey }) => {
+export const DeleteKey: React.FC<Props> = ({ apiKey, keyAuthId }) => {
   const router = useRouter();
   const [open, setOpen] = useState(false);
 
   const deleteKey = trpc.key.delete.useMutation({
     onSuccess() {
+      revalidate(`/keys/${keyAuthId}`);
```
Comment (chronark) [inline apps/dashboard/app/(app)/keys/[keyAuthId]/[keyId]/settings/delete-key.tsx:35]: why is this necessary?

File: apps/dashboard/app/(app)/keys/[keyAuthId]/keys.tsx [hunk 6]
```diff
@@ -9,6 +9,8 @@ type Props = {
   keyAuthId: string;
 };
 
+export const dynamic = "force-dynamic";
+export const revalidate = true;
```
Comment (chronark) [inline apps/dashboard/app/(app)/keys/[keyAuthId]/keys.tsx:13]: what does this do?

**Comment by perkinsjr** [comment]
Absolutely :+1:

**Review by chronark — APPROVED** [review]
LGTM if you have verified by testing it in the UI

**Comment by chronark** [comment]
fyi this is blocked cause you had a “test” change request @perkinsjr can you approve or revoke that

**Review by perkinsjr — APPROVED** [review]

**Comment by MichaelUnkey** [comment]
I did test it. But I also did last time and then it was broken for James

**Pull request merged**